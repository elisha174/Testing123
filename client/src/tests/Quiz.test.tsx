import pretty from 'pretty';
import { render } from '@testing-library/react';
import Quiz from '../components/Quiz';

describe('Quiz', () => {
  it('should contain the expected text', () => {
    render(<Quiz />);

    const itemElement = document.querySelector('h2');

    if (itemElement) {
      expect(itemElement.textContent).toBe('Quiz Completed');  
    }
    
  });

  it('should match snapshot', () => {
    render(<Quiz />);

    const calculatorElement = document.querySelector('.p-4 text-center');

    if (calculatorElement) {
      expect(pretty(calculatorElement.innerHTML)).toMatchSnapshot();
    }
  });
});
