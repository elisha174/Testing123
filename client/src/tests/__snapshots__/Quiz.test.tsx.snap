// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Calculator > should match snapshot 1`] = `
"<button className="btn btn-primary d-inline-block mx-auto" onClick={handleStartQuiz}>
          Start Quiz
        </button>"
`;
